### PR TITLE
Address is now defined as a simple string

### PR DESCRIPTION
**Pull request by andreibancioiu**
Address is now a simple string.

Functions of the previous definitions of `Address` and  `AddressFactory`  have been absorbed into `AddressComputer`.

### CODE REVIEW

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: sdk-core/address.md [hunk 1]
```diff
@@ -1,99 +1,92 @@
...
-    // Name should be adjusted to match the language's convention.
-    to_bech32(): string;
-
-    // Returns the address as a string (hex).
-    // Name should be adjusted to match the language's convention.
-    to_hex(): string;
-
-    // Returns the underlying public key.
-    get_public_key(): bytes;
+Generally speaking, the `Address` type should be a `string` in all implementations.
```
[the remark this answers is not in the document]
Comment (andrei-marinica) [reply sdk-core/address.md:3]: In mandos we use the value interpreter to express things in textual form:
`bech32:erd123445......`
`address:dummy_address`
`0x1234aaa...`

In Rust I want to create string wrappers that have the same function: making an address very readable, while also specifying the exact format
`Bech32("erd123445......")`
`TestAddress"dummy_address")`
`Hex("0x1234aaa...")`

I think making the exact format used obvious is essential.

[the remark this answers is not in the document]
Comment (andrei-marinica) [reply sdk-core/address.md:3]: Can't we also write in these languages something in the spirit of newtype? An object that wraps the string, while also giving some additional context?

**Comment by andreibancioiu** [comment]
Change is abandoned. Components will only be slightly adjusted, but in a different PR.

File: sdk-core/address.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply sdk-core/address.md:3]: @andrei-marinica that is why we had the IAddress definition. A struct that made the conversions automatically so the confusion between the string types was eliminated.

**Pull request closed without merging**